backend/assessments/1/src/main/java/org/example/IPLApplication2.java

```diff
@@ -0,0 +1,271 @@
+                    String teamName = scanner.nextLine();
+                    Logging.logInfo("Enter minimum wickets: ");
+                    int minWickets = scanner.nextInt();
+                    List<Player> bowlers = findBowlersWithMinimumWickets(teamName, minWickets);
+                    Logging.logInfo("Bowlers with at least " + minWickets + " wickets for " + teamName + ":");
+                    for (Player bowler : bowlers) {
+                        Logging.logInfo(bowler.getName() + " - " + bowler.getWickets() + " wickets");
+                    }
+                    break;
+                case 2:
+                    Logging.logInfo("Enter team name: ");
+                    teamName = scanner.nextLine();
+                    displayHighestWicketTakerAndRunScorer(teamName);
+                    break;
+                case 3:
+                    displayTop3RunScorersAndWicketTakers();
+                    break;
+                case 4:
+                    generateMatchFixtures();
+                    break;
+                case 5:
+                    Logging.logInfo("Exiting the application. Goodbye!");
+                    break;
+                default:
+                    Logging.logInfo("Invalid choice. Please enter a valid option.");
+            }
+        } while (choice != 5);
+    }
+
+    private static void loadDataFromCSV(String filePath) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+           reader.readLine();
+
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split(",");
+                String name = data[0];
+                String team = data[1];
+                String role = data[2];
+                int matches = Integer.parseInt(data[3]);
+                int runs = Integer.parseInt(data[4]);
+                double average = Double.parseDouble(data[5]);
+                double strikeRate = Double.parseDouble(data[6]);
+                int wickets = Integer.parseInt(data[7]);
+
+                Player player = new Player(name, team, role, matches, runs, average, strikeRate, wickets);
+                addToTeam(player);
+            }
+        } catch (IOException | NumberFormatException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void addToTeam(Player player) {
+        boolean teamExists = false;
+        for (Team team : teams) {
+            if (team.getTeamName().equalsIgnoreCase(player.getTeam())) {
+                teamExists = true;
+                team.addPlayer(player);
+                break;
+            }
+        }
+
+        if (!teamExists) {
+            Team newTeam = new Team(player.getTeam());
+            newTeam.addPlayer(player);
+            teams.add(newTeam);
+        }
+    }
+
+    private static List<Player> findBowlersWithMinimumWickets(String teamName, int minWickets) {
```

Comment:
Add validations for all methods. And abstract them out.
Wrap them in a validator wrapper/function.